fabtests/ubertest/uber.c

```diff
@@ -330,298 +329,127 @@ static int ft_recv_test_info(void)
 	return 0;
 }
 
-static int ft_exchange_uint32(uint32_t local, uint32_t *remote)
+static int ft_send_result(int err, struct fi_info *info)
 {
-	uint32_t local_net = htonl(local);
 	int ret;
-
-	ret = ft_sock_send(sock, &local_net, sizeof local);
+	ret = ft_sock_send(sock, &err, sizeof err);
 	if (ret) {
 		FT_PRINTERR("ft_sock_send", ret);
 		return ret;
 	}
-
-	ret = ft_sock_recv(sock, remote, sizeof *remote);
-	if (ret) {
-		FT_PRINTERR("ft_sock_recv", ret);
-		return ret;
+	if (err) {
+		printf("Ending test %d, result: %s\n", test_info.test_index,
+			fi_strerror(-err));
+		return err;
 	}
 
-	*remote = ntohl(*remote);
-
 	return 0;
 }
 
-static int ft_skip_info(struct fi_info *hints, struct fi_info *info)
+static int ft_recv_result(struct fi_info *info)
 {
-	uint32_t remote_protocol, skip, remote_skip;
-	size_t len;
-	int ret;
-
-	//make sure remote side is using the same protocol
-	ret = ft_exchange_uint32(info->ep_attr->protocol, &remote_protocol);
-	if (ret)
-		return ret;
-
-	if (info->ep_attr->protocol != remote_protocol)
-		return 1;
-
-	//check needed to skip utility providers, unless requested
-	skip = (!ft_util_name(hints->fabric_attr->prov_name, &len) &&
-		strcasecmp(hints->fabric_attr->prov_name,
-		info->fabric_attr->prov_name));
-
-	ret = ft_exchange_uint32(skip, &remote_skip);
-	if (ret)
+	int ret, err = 0;
+	ret = ft_sock_recv(sock, &err, sizeof err);
+	if (ret) {
+		FT_PRINTERR("ft_sock_recv", ret);
 		return ret;
+	}
+	if (err) {
+		printf("Ending test %d, result: %s\n", test_info.test_index,
+			fi_strerror(-err));
+	}
 
-	return skip || remote_skip;
+	return err;
 }
 
-static int ft_transfer_subindex(int subindex, int *remote_idx)
+static int ft_server_setup(struct fi_info *hints, struct fi_info *info)
 {
-	int ret;
+	int ret = 0;
```

Comment:
Drop the assignment to 0 here.  See comments below to use a common error exit path.